### ISSUE TITLE
Cursor (blit?) regression in 2.0.16

### ISSUE THREAD

**Issue by pushfoo** [pyglet/pyglet#1169]
## Describe the bug

**TL;DR:** https://github.com/pyglet/pyglet/pull/1109 may have done something weird to blending in 2.0.16

I'm filing this on behalf of a Discord user I'm guiding through getting diagnostic info. [Discord discussion here](https://discord.com/channels/438622377094414346/463526340168122368/1262721983276122132).

### Expected Behavior
![image](https://github.com/user-attachments/assets/e11a7644-7a3c-46b3-8a61-28cd6d00b0b6)

### Actual Behavior 
![image](https://github.com/user-attachments/assets/349b3aa9-54eb-46cf-a0a0-3b5d3eddec26)

## How to Reproduce
Image:
![image](https://github.com/user-attachments/assets/22a8eae8-3431-4cba-b3e5-782f313c7ffe)

```py
import pyglet

win = pyglet.window.Window()
image = pyglet.image.load('cursor.png')

@win.event
def on_draw():
    win.clear()
    image.get_texture().blit(5,5)

pyglet.app.run()
```

## System Info

> Windows 10 64-bit

[full `python -m pyglet.info`](https://gist.github.com/pushfoo/1182451dd62144a3dcaa587f3de1157e)

**Comment by caffeinepills** [comment]
This should be resolved with the latest PR.
